rbd/pool_nautilus.go

```diff
@@ -26,6 +26,9 @@ func GetPoolMetadata(ioctx *rados.IOContext, key string) (string, error) {
 	cKey := C.CString(key)
 	defer C.free(unsafe.Pointer(cKey))
 
+	if ioctx == nil {
```

Comment:
I prefer error checking to be done as early as possible. Also check `key == ""`?

Reply:
Will move it up.
For key == "", I have added test cases which prove that ceph functions handles gracefully if key is "" and we don't have to explicitly do it in go-ceph functions.

Reply:
I agree to moving the check up. I also agree with not adding lots of redundant checks but just validate ceph calls do the right thing.